Chatkit/Chatkit/Chatkit.swift

```diff
@@ -289,6 +289,26 @@ public class Chatkit {
         }
     }
     
+    /// Retrieves a static snapshot of the `User`s who are currently members of the `Room`.
+    ///
+    /// - Parameters:
+    ///     - `room`: The `Room` for which members should be retrieved.
+    public func members(for room: Room) -> [User] {
+        let roomManagedObjectID = room.objectID
+        
+        let context = persistenceController.mainContext
+        let predicate = NSPredicate(format: "ANY %K == %@", #keyPath(UserEntity.room), roomManagedObjectID)
+        let sortDescriptor = NSSortDescriptor(key: #keyPath(UserEntity.identifier), ascending: true) { (lhs, rhs) -> ComparisonResult in
+            guard let lhsString = lhs as? String, let lhs = Int(lhsString), let rhsString = rhs as? String, let rhs = Int(rhsString) else {
+                return .orderedSame
+            }
+            
+            return NSNumber(value: lhs).compare(NSNumber(value: rhs))
+        }
+
+        let memberEntities = context.fetchAll(UserEntity.self, withRelationships: nil, limit: 999, sortedBy: [sortDescriptor], filteredBy: predicate)
```

Comment:
There is one problem here. This code assumes that the method will be always called from the main thread, because `mainContext` is confined to that thread. That might not always be the case. In order to fix that we can wrap this code into `performAndWait` closure.

Reply:
Cool. I did wonder when I wrote it. It's a shame that `performAndWait` returns `()` rather than a generic. 🤷‍♂